rstudio/rstudio
# Add fr translations rosetta install warning

### Intent

Addresses https://github.com/rstudio/rstudio/issues/13548

#### Preview
<img width="1728" alt="image" src="https://github.com/rstudio/rstudio/assets/25834218/533d7d26-80b8-481a-a236-916ac1e29520">


### Approach
- add English and French strings for the detect rosetta dialog text
- relocate the `detectRosetta` call from the session launcher to the initialization of the desktop header, so that the i18n framework is initialized and set to the correct language by the time we call `detectRosetta`
    - one tradeoff of this approach is that the IDE window shows up in a loading state behind the Detect Rosetta dialog, where previously the IDE window would not open until the user selects a choice from the Detect Rosetta dialog
- removed a call to initialize i18next from the detect-rosetta code, which was interfering with i18n initialization and language detection

#### Sidenote
We seem to be using i18next incorrectly (?) based on [the docs](https://www.i18next.com/overview/api):
> The default export of the i18next module is an i18next instance ready to be initialized by calling init. You can create additional instances using the [createInstance](https://github.com/overview/api#createinstance) function.
>
> Do not call init multiple times.
> To change language use [changeLanguage](https://github.com/overview/api#changelanguage). If you need complete different configs use [createInstance](https://github.com/overview/api#createinstance) or [cloneInstance](https://github.com/overview/api#cloneinstance).

We call `i18next.init` in several places, but the docs instead suggest to create separate instances if we need a different config, or call `init` once and wait until it initialized. I also tried using `i18next.on('initialized', callback)`, but my callback wasn't getting called. Maybe has to do with the multiple calls to `i18next.init` confusing the listener?

### Automated Tests

none

### QA Notes

Same testing notes as https://github.com/rstudio/rstudio/pull/13518, with an added step of changing the language to French in global settings, then restarting the IDE to ensure you are prompted to install rosetta in French.

Note that this dialog should not appear on Linux or Windows, since Rosetta is Mac-only. I have tested in a Linux Ubuntu 22.04 VM that the check does not occur.

### Documentation
none

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` already covered in https://github.com/rstudio/rstudio/pull/13518
